src/OrchardCore.Modules/OrchardCore.Markdown/Handlers/MarkdownBodyPartHandler.cs

```diff
@@ -25,12 +24,6 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
         {
             return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
-                if (_bodyAspect != null)
-                {
-                    bodyAspect.Body = _bodyAspect;
-                    return;
-                }
-
```

Comment:
yes, this is cached for the request, so that we are not having to generate the markdown/liquid rendering multiple times on the same request.

What you could do is move the cached body aspect to a `Dictionary<string, string>()` keyed by `ContentItemId`

Also there are other instances of this usage, for the `HtmlBodyPartHandler` (and others, you will find) which would want to be updated as well.

Reply:
Ya, I was thinking about other handlers .. I will try to update the PR by introducing a dictionary cache as you suggested 

Thanks